src/AbstractFixer.php

```diff
@@ -34,7 +43,7 @@ public function __construct()
     {
         if ($this instanceof ConfigurableFixerInterface) {
             try {
-                $this->configure(null);
+                $this->configure(array());
```

Comment:
I agree and tried to do that but I faced two issues.

First, it has unexpected results with configuration definitions that depend on some context of the fixer. It made `header_comment` configuration throw errors because it depends on the `WhitespacesFixerConfig` instance of the fixer, which can change between `configure()` calls. It can be fixed by forcing this fixer to always create a new configuration definition instance instead of reusing the first one.

But the second issue is an increasing of memory usage: I wasn't able to run the test suite without exceeding the memory limit on my setup. The limit is the default 128M so it's not high but the test suite usually requires ~40M. I'll have to investigate more to determine if this memory issue is a direct consequence of this change or something else that requires some optimization first. It looks like this only concerns the test suite though, as running the tool on the project didn't show a noticeable change in the consumed memory.

Anyway, applying this change didn't seem to have an impact on performance either, so I'm not sure it's worth it.